src/App.jsx

```diff
@@ -14,70 +16,66 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [good, selectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {good === null ? (
+          'No goods selected'
+        ) : (
+          <>
+            {`${good} is selected`}
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
             <button
-              data-cy="RemoveButton"
+              data-cy="ClearButton"
               type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+              className="delete ml-3"
+              onClick={() => {
+                selectedGood(null);
+              }}
+            />
+          </>
+        )}
+      </h1>
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
+      <table className="table">
+        <tbody>
+          {goods.map(item => (
+            <tr
+              data-cy="Good"
+              key={item}
+              className={classNames({
+                'has-background-success-light': good === item,
+              })}
             >
-              +
-            </button>
-          </td>
+              <td>
+                <button
+                  data-cy={good === item ? 'RemoveButton' : 'AddButton'}
+                  className={classNames('button', {
+                    'is-info': good === item,
+                  })}
+                  type="button"
+                  onClick={() => {
+                    if (good === item) {
+                      selectedGood(null);
+                    } else {
+                      selectedGood(item);
+                    }
+                  }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)